corehq/apps/api/resources/v0_5.py

```diff
@@ -1295,3 +1319,104 @@ def _get_compound_filter(cls, param_field_name: str, params: NavigationEventAudi
                 filter_obj = cls.COMPOUND_FILTERS[param_field_name](qualifier, val)
                 compound_filter &= Q(**filter_obj)
         return compound_filter
+
+
+@csrf_exempt
+@allow_cors(['GET'])
+@api_auth()
+@require_can_edit_data
+@requires_privilege_with_fallback(privileges.API_ACCESS)
+@api_throttle
+def get_ucr_data(request, domain):
+    if not toggles.EXPORT_DATA_SOURCE_DATA.enabled(domain):
+        return HttpResponseForbidden()
+    try:
+        if request.method == 'GET':
+            config_id = request.GET.get("data_source_id")
+            if not config_id:
+                return HttpResponseBadRequest("Missing data_source_id parameter")
+            return get_datasource_data(request, config_id, domain)
+        return JsonResponse({'error': "Request method not allowed"}, status=405)
+    except BadRequest as e:
+        return JsonResponse({'error': str(e)}, status=400)
+
+
+def get_datasource_data(request, config_id, domain):
+    """Fetch data of the datasource specified by `config_id` in a paginated manner"""
+    config, _ = get_datasource_config(config_id, domain)
+    datasource_adapter = get_indicator_adapter(config, load_source='export_data_source')
+    cursor_params = get_request_params(request).params
```

Comment:
I moved some of these function to other files where I thought it might belong. https://github.com/dimagi/commcare-hq/pull/33380/commits/b6cfc2a6eeec0102a6e804e9aa098d80a8d3badf